src/replayq_registry.erl

```diff
@@ -0,0 +1,86 @@
+-module(replayq_registry).
+
+-behaviour(gen_server).
+
+%% API
+-export([
+    start_link/0,
+
+    register_committer/2
+]).
+
+%% `gen_server' API
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2
+]).
+
+%%------------------------------------------------------------------------------
+%% Type declarations
+%%------------------------------------------------------------------------------
+
+%% call/cast/info events
+-record(register_committer, {dir :: filename:filename_all(), pid :: pid()}).
+
+%%------------------------------------------------------------------------------
+%% API
+%%------------------------------------------------------------------------------
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+register_committer(Dir, Pid) ->
+    gen_server:call(?MODULE, #register_committer{dir = Dir, pid = Pid}, infinity).
+
+%%------------------------------------------------------------------------------
+%% `gen_server' API
+%%------------------------------------------------------------------------------
+
+init(_Opts) ->
+    process_flag(trap_exit, true),
+    State = #{committers => #{}},
+    {ok, State}.
+
+handle_call(#register_committer{dir = Dir, pid = Pid}, _From, State0) ->
+    %% Should we expand the directory path to avoid tricks with links and relative paths?
+    #{committers := Committers0} = State0,
+    case Committers0 of
+        #{Dir := Pid} ->
+            %% Already registered to the same pid.
```

Comment:
this should not be ok, because it means the same process called open twice without closing first.

Reply:
forgot the pid is actually the committer pid.
then the question is: is this possible to happen ?

Reply:
no, in the current implementation.

thinking of this as a general registry, repeating the exact same registration should be idempotent.

Reply:
after we discussed, I changed the last commit to attempt to register the called of `open` as soon as possible to avoid side-effects.